lib/rules/mixin-name-format.js

```diff
@@ -0,0 +1,78 @@
+// Note that this file is nearly identical to function-name-format.js, variable-name-format.js, and placeholder-name-format.js
+'use strict';
+
+var helpers = require('../helpers');
+
+module.exports = {
+  'name': 'mixin-name-format',
+  'defaults': {
+    'allow-leading-underscore': true,
+    'convention': 'hyphenatedlowercase'
+  },
+  'detect': function (ast, parser) {
+    var result = [];
+
+    ast.traverseByTypes(['mixin', 'include'], function (node) {
+      var name;
+
+      if (node.type === 'mixin') {
+        name = node.first('ident').content;
+      }
+      else {
+        name = node.first('simpleSelector').first('ident').content;
+      }
+
+      var originalName = name;
```

Comment:
Updated. I didn't like the variable `originalName` very much so I went with `name` and `strippedName`.